### ISSUE TITLE
enshittification / needing an account

### ISSUE THREAD

**Issue by ChristianSch** [getinsomnia/insomnia#6577]
### Expected Behavior

I would like to be able to use insomnia without an account or be warned BEFORE an update that you'll lock me out of my stuff. 

### Actual Behavior

you just locked me out

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

I really really do know that this is not the place, but wtf guys, come on? Have you learned nothing from Postman? Their enshittification gave you the traffic to grow in the first place and now you're doing the same shit?

### Insomnia Version

v8.0.0

### What operating system are you using?

macOS

### Operating System Version

macOS sonoma

### Installation method

download

### Last Known Working Insomnia version

< 8.0.0

**Comment by imitko** [comment]
I have same problem too, all my projects was vanished, even worse the login tell me to use  password for old account and no even way to reset it.

**Comment by elevesque-nexapp** [comment]
Same issue here, I restarted to apply what appeared to be a regular update and now I apparently don't have access to my projects!? Needless to say I'm quite frustrated.

The whole reason I used Insomnia was because it wasn't trying to push to create an account. If that's the way things are going to be I'll just return to Postman.

**Comment by erebor** [comment]
Same here.

**Comment by bholm** [comment]
For anyone affected, you can downgrade to [2023.5.8](https://github.com/Kong/insomnia/releases/tag/core%402023.5.8) and get back all your projects for now but I guess I'll need to start investigating alternatives now. I am very disappointed with this move.

**Comment by alexandrudumencu** [comment]
Same here.
At least let me export the requests before locking me out.
I tried setting up an account via GitHub, but it fails.

**Comment by jeffwalter-rum** [comment]
Time to fork.

**Comment by lfleal** [comment]
Same here. I was in the middle of testing when I was prompted to restart, then I lost access to everything. I turn off auto updated and downgraded to last release. I confirmed that auto update is still off. I have access to my projects again. I will be migrating off of Insomnia to another solution. Same here I used Postman and migrated here for the same reasons. 

For those that just got locked out. 1) turn off update 2) downgrade to 2023.5.7

**Comment by mkdillard** [comment]
Yeah, I have to agree here, this was completely unexpected.  I clicked though to the scratchpad to avoid the account sign up and all my stuff was gone?  If you can copy the data to associate with an account online you can copy the data to keep it local and anonymous.  This was an incredibly short sighted decision to put folks in this position, especially without any warning that local collections would likewise be removed/hidden.

I was excited to have Insomnia as a Postman alternative where I could keep all my stuff local and not in the cloud, but now I guess I may have to start looking again.

**Comment by erebor** [comment]
Good reminder to export data while you can:

<img width="696" alt="Screenshot of Insomnia (9-27-23, 2-33-19 PM)" src="https://github.com/Kong/insomnia/assets/4887/652f41d0-be25-4b63-95ef-804f4419a5fa">

**Comment by geoshak** [comment]
Same problem here.

**Comment by petehl** [comment]
> For anyone affected, you can downgrade to [2023.5.8](https://github.com/Kong/insomnia/releases/tag/core%402023.5.8) and get back all your projects for now but I guess I'll need to start investigating alternatives now. I am very disappointed with this move.

What great incentive to turn off updates, and begin to look for a simpler alternative.

**Comment by subnetmarco** [comment]
More info [here](https://github.com/Kong/insomnia/issues/6577#issuecomment-1738131554).

The data can be optionally exported before continuing with the cloud synchronization. The data can also be exported after logging in. 

The data is fully end-to-end encrypted (E2EE) in the cloud for all users, nobody but you can access it, which is why we request a passphrase and why losing the passphrase will cause the data to be lost.

With Git sync the data never goes to the cloud but stays in the Git repository.

The challenge is that we cannot continue to offer a great product with all these capabilities while working on an abstraction layer that supports two different storage backends for all data in Insomnia, adds too much complexity to the whole product. Unlike Postman, the data is end to end encrypted, so there is data sharing concern with Insomnia.

**Comment by smnbbrv** [comment]
@subnetmarco that's what should be told before the upgrade is automatically applied, shouldn't it?

**Comment by mkdillard** [comment]
> The challenge is that we cannot continue to offer a great product with all these capabilities while working on an abstraction layer that supports two different storage backends for all data in Insomnia, adds too much complexity to the whole product.

Yet 8.0 appears to offer an offline anonymous scratchpad and your update didn't offer an option to migrate existing local collection to that, just the cloud?  So local data storage is being offered or not?

**Comment by ChristianSch** [comment]
@subnetmarco  You absolutely knew what this would cause and you absolutely know that there will be commotion and people will move on. You just hope that enough stick around for money to be made. Being a CTO myself I get the business case. But you lost a lot of credibility today. So much so that (imho) this hurts Kong business standing. How can I trust you folks? How do I know that you are a reliable partner and support me and my developers in our work? Instead you pull the same shit Postman did. We'll be looking into alternatives.

**Comment by subnetmarco** [comment]
@smnbbrv it was told, there is a wizard that shows up that tells that the data will be synchronized to the cloud and even give an opportunity to export it.

**Comment by mkdillard** [comment]
You mean the wizard that didn't appear until after the update was accepted?

**Comment by bobdanielson** [comment]
@subnetmarco My employer has explicit policies against storing our data in someone else's cloud.  Even if the data is encrypted since we cannot audit it for compliance it will not be allowed.  Insomnia is going to be completely blocked even for individual license holders here in very short order with this update.

**Comment by subnetmarco** [comment]
@mkdillard the Scratch Pad is heavily limited in capability because of this, which is not the experience we want to give to Insomnia users moving forward. The whole storage backend being local also caused data migration issues pretty much since Insomnia was born. 

If your is a privacy concern, the data is end-to-end encrypted on every user account, including the Free ones.

**Comment by subnetmarco** [comment]
@mkdillard the wizard that shows up after updating the Insomnia binary doesn't synchronize the data until after continuing, and it shows an export capability to export the data locally before doing so.

**Comment by subnetmarco** [comment]
For anybody where this is a problem: you can download an old version of Insomnia and also export the data from there. The data is **not** lost.

**Comment by mkdillard** [comment]
> the wizard that shows up after updating the Insomnia binary

That's the point though, the wizard that explains what's changed and where our data will be going didn't show up until after the update.  We had no way of knowing that 8.0 would be such a drastic change of direction before clicking to apply the update.

As an aside, my feedback as a user is:
My number 1 criteria for choosing Insomnia was the local offline collections, I came here from Postman after they pulled the same stunt forcing everyone to cloud.  It's clear that's not a use case you're interested in supporting, and as nice as Insomnia has been, it's no longer a viable choice for my REST client needs.

The way you all surprised everyone with this update suddenly (at least I haven't seen communication about this huge change to data storage) also leaves a bad taste in the mouth.

**Comment by ClickyCoyote** [comment]
> The data can be optionally exported before continuing with the cloud synchronization. The data can also be exported after logging in.
> 
> The data is fully end-to-end encrypted (E2EE) in the cloud for all users, nobody but you can access it, which is why we request a passphrase and why losing the passphrase will cause the data to be lost.
> 
> With Git sync the data never goes to the cloud but stays in the Git repository.
> 
> The challenge is that we cannot continue to offer a great product with all these capabilities while working on an abstraction layer that supports two different storage backends for all data in Insomnia, adds too much complexity to the whole product. Unlike Postman, the data is end to end encrypted, so there is data sharing concern with Insomnia.

I was never given an option to export, I was presented with the login screen immediately upon upgrading. Additionally, I even tried logging in via GitHub and after a few 500 errors, got to a screen showing my account  was being migrated but nothing ever happened.

**Comment by smnbbrv** [comment]
>  @smnbbrv it was told, there is a wizard that shows up that tells that the data will be synchronized to the cloud and even give an opportunity to export it.

@subnetmarco not really. I got a popup - version 8 is ready and I need to restart. Everything else happened after the upgrade. What I have now is empty scratchpad with no option to proceed

**Comment by AlaricWhitney** [comment]
I switched away from Postman due to the same stunt.  It appears that I will have to look for an alternative now that all of my collections are gone and being held hostage without any warning.

**Comment by bobdanielson** [comment]
> > The data can be optionally exported before continuing with the cloud synchronization. The data can also be exported after logging in.
> > The data is fully end-to-end encrypted (E2EE) in the cloud for all users, nobody but you can access it, which is why we request a passphrase and why losing the passphrase will cause the data to be lost.
> > With Git sync the data never goes to the cloud but stays in the Git repository.
> > The challenge is that we cannot continue to offer a great product with all these capabilities while working on an abstraction layer that supports two different storage backends for all data in Insomnia, adds too much complexity to the whole product. Unlike Postman, the data is end to end encrypted, so there is data sharing concern with Insomnia.
> 
> I was never given an option to export, I was presented with the login screen immediately upon upgrading. Additionally, I even tried logging in via GitHub and after a few 500 errors, got to a screen showing my account was being migrated but nothing ever happened.

Worse yet I now know of multiple people who had closed Insomnia without accepting the update, but because auto-updates was on their client restarted and they were immediately on version 8 and blocked by the cloud account dialog which they are not allowed by policy to use.  The only recourse is to downgrade and disable updates.  Many are already throwing their hands up in the air and looking for compliant alternatives.

**Comment by javierpaez** [comment]
same issue 👎

**Comment by crestonbunch** [comment]
I went to Insomnia because I couldn't live with Postman. Where do I go now?

**Comment by subnetmarco** [comment]
I want to point out this:

1) The data is never migrated to the cloud unless you create an account or login. It's still local.

2) The data can be exported with an older version of Insomnia or it can be exported during the onboarding wizard that we added for Insomnia 8.0. The collection can also be re-imported into Scratch Pad, which will keep the data of the collection forever local without requiring an account.

![image](https://github.com/Kong/insomnia/assets/964813/58a8576f-5705-485f-8836-78b39fcc4bc2)

**Comment by crestonbunch** [comment]
> The data is never migrated to the cloud unless you create an account or login. It's still local.

The only option to remain local is to use the "scratchpad". The scratch pad does not contain previous data. The update _completely_ breaks existing workflows if you do not create an account. It is unnacceptable.

**Comment by codeadict** [comment]
Welp back to curl folks

**Comment by clayton-pierce** [comment]
I was thinking their nag to update was very pervasive. Now it makes sense why.

**Comment by pawel-kuras** [comment]
need link of the fork we plan to move over to.

**Comment by artiebob** [comment]
> Time to fork.

If someone wants to create a product that doesn't store your data externally I know many people willing to pay for it.

**Comment by wreppun** [comment]
> it can be exported during the onboarding wizard that we added for Insomnia 8.0

I mean, if you think that tiny bolded link (which disappears, forever, once you hit continue) is an adequate path for users to take, that explains a lot about how this whole fiasco happened in the first place.  

"hiring, ~~thinking~~, coding"

**Comment by AlaricWhitney** [comment]
> I want to point out this:
> 
> 1. The data is never migrated to the cloud unless you create an account or login. It's still local.
> 2. The data can be exported with an older version of Insomnia or it can be exported during the onboarding wizard that we added for Insomnia 8.0. The collection can also be re-imported into Scratch Pad, which will keep the data of the collection forever local without requiring an account.
> 
> 

What's flawed about the logic is that there is no warning about this prior to the upgrade, nor is there an option to export the data prior to finishing the update, so from a user experience this process is shocking.  There is the implied assumption that we are now forced on our own to find, download, and install the previous version to recover access to our data.  At no point during the installation of the update we were given the option to export our data before the upgrade, or warned that we would loose access to our data with the upgrade.

For those interested, here's link to the previous version so we can gain access to our data again: https://github.com/Kong/insomnia/releases/tag/core%402023.5.8

**Comment by RayJSeth** [comment]
> > Time to fork.
> 
> If someone wants to create a product that doesn't store your data externally I know many people willing to pay for it.

Not affiliated with any of these projects, just anecdote.
My journey has been Postman => Insomnia => Restfox.
Just made that third jump today and everything ported over fine and looks to be working great. 🤷

**Comment by erraggy** [comment]
What an amazing introduction to Kong's product offerings! I went from a happy user of Insomnia that was sharing how great it was with all of my colleagues, to a confused and upset online comment writer in just one short week!
🥳 

Interestingly enough, I had just migrated to a new laptop and was overjoyed yesterday evening when I completed my development environment and closed my laptop for the night. I opened my laptop today to find that Insomnia has Amnesia and could not locate my data.

Needless to say... 

N O T   A   G O O D   L O O K

👎🏻 👎🏻

**Comment by HauLuu** [comment]
> I want to point out this:
> 
> 1. The data is never migrated to the cloud unless you create an account or login. It's still local.
> 2. The data can be exported with an older version of Insomnia or it can be exported during the onboarding wizard that we added for Insomnia 8.0. The collection can also be re-imported into Scratch Pad, which will keep the data of the collection forever local without requiring an account.
> 
> ![image](https://user-images.githubusercontent.com/964813/271105592-58a8576f-5705-485f-8836-78b39fcc4bc2.png)

Why did they make the warning message way too small on the wizard?

Why don't let us use the Scratchpad with old data available?

It's time to move.

**Comment by crestonbunch** [comment]
I have evangelized Insomnia at my company for many years because I found the experience to be superior to Postman. No longer will I continue. I am now actively discussing alternatives with engineers who have been burned by this.

**Comment by lucacri** [comment]
This is horrible move that inspire **zero** confidence in Kong & Insomnia. I just lost 30 mins of work because I updated and needed to figure out why everything was missing, and I'm not uploading any of my API calls to a cloud provider, specially one that just pulled a stunt like this one.

Time to find an alternative, or fork this

**Comment by ryber** [comment]
There are security departments at large corporations across the world right now trying to figure out how to block their developers from creating accounts and inadvertently uploading all their oauth secrets to someone else's cloud.

**Comment by crestonbunch** [comment]
> The whole storage backend being local also caused data migration issues pretty much since Insomnia was born.

This is so dumb. Syncing is a solved problem. You can do it with iCloud. You can do it with Dropbox. You can do it with rsync, a thumbdrive, a pigeon carrying a thumb drive, honestly anything. Insomnia doesn't need to be in the business of syncing files. Just give us an HTTP client and stay out of our way.

**Comment by rtgoodwin** [comment]
Complete failure here as well, and deleted the app as I can no longer trust it will only store locally.

**Comment by dkapellusch** [comment]
Yeah this is absurd. I've always been a huge proponent of Insomnia over alternatives like PostMan. I can't believe anyone thought this was a good idea. 

I can't even recover my collections without downgrading, instant uninstall.

**Comment by bobdanielson** [comment]
I'm watching the fork count slowly climb...

**Comment by ljohnston** [comment]
This is almost funny. I _just_ migrated to Insomnia from Postman earlier this week because Postman finally enforced the "let's make them all go to the cloud thing". And now this just two days later. Thank goodness I hadn't put much of anything into Insomnia yet!

I work for one of those companies that will **_never, ever, ever_** allow us to store this kind of stuff in the cloud. Postman is now off of the approved software list due to their recent changes. Thought Insomnia might be worth looking at as an alternative. Not now.

All of the above aside, this was an unbelievable bush league move. The way it was **_totally_** a blindside is unconscionable.

**Comment by MaxLithia** [comment]
Huge disappointment. Very blatant business strategy not backed by the technical justification.

**Comment by williamyeny** [comment]
NOT a good move. Most devs have sensitive testing info that should stay local. I would not be comfortable with my data uploaded to "Insomnia Cloud", even if it is encrypted.

**Comment by billriess** [comment]
Ignoring the fact that everything is cloud-based now (which sucks) my team cannot access anything. When they login it just takes them to the payment screen asking them to upgrade their accounts. They are already part of a paid team organization and their emails are in the list. They have no option to do anything but fill out payment details or logout. So at this point we are paying for an unusable service.

**Comment by subnetmarco** [comment]
@billriess they can continue with "Free" and still be able to access the team.

**Comment by billriess** [comment]
> @billriess they can continue with "Free" and still be able to access the team.

This worked. Thank you! It's so obscure and makes no sense that they need to click "Downgrade" their Team Account to Free Account in order to access the paid Team Account...

**Comment by subnetmarco** [comment]
@billriess it's a little bit like GitHub: the users can individually be on a Free plan, but when they are invited to join an organization and work on the organization projects, they are working under the umbrella of the organization subscription (Team, in your case).

**Comment by billriess** [comment]
@subnetmarco I get that but this should have happened automatically in the background when the app was migrated to this new system. Again, its very confusing to ask someone to downgrade their account to a free plan from a team plan in order for them to access the team plan they already are a member of. At the very least, it should allow them access to the team plan they are already a member of and they can setup their personal account later.

Even more confusing, the page says "Your subscription is past due" which would make the user think there is an issue with the billing on the organization. This entire flow is bad.

**Comment by Notuom** [comment]
This is anti-user design trying to get people onto their account system to sell licenses. I downgraded to the previous version and am now actively looking for different software. I moved away from Postman a few years ago because it had become bloated with the same anti-user design. As many others here, now looking for alternatives.

**Comment by subnetmarco** [comment]
Given the large amount of feedback, I am writing here the current situation.

## What changed with Insomnia 8.0 when it comes to your data?

With Insomnia 8.0 we are doing two new things:

1. After downloading the new binary (either manually or via the auto-update) we show an onboarding wizard that communicates that your data (collections, tests, etc) will be synchronized to the cloud after the user creates an account or signs in. There is an option to export the data, and the data is not actually synchronized until either an account is being created, or a login is being performed.
2. We are also introducing a new Scratch Pad mode that is always local and never synchronized to the cloud, and doesn't require an account with Insomnia.

The data is always end-to-end encrypted (E2EE) in the cloud, therefore no-one (not even Insomnia or Kong) can access it without a valid passphrase. Which is also why losing the passphrase can lead to loss of data, it's because of E2EE.

## Can Insomnia still be used for Free?

Yes, we offer a Free plan that you can use.

## Why were these changes introduced?

These changes were introduced because:

1. We have a significant amount of the user base that wants to collaborate and automatically synchronize their data across their clients. This is impossible without leveraging the cloud to do so. Insomnia uses E2EE for each user, including the Free ones, to still be able to do this securely.
2. There is a significant effort in maintaining full support for an abstraction that supports two different backend (in the cloud, and locally), which is causing a large amount of issues and migration problems. We are therefore introducing limited support for the local backend storage in Scratch Pad only (more limited than the cloud one), and investing in making the cloud storage solid, stable and secure.

## How to retrieve your data?

1. Your data can be retrieved in Insomnia 8.0 by exporting it via the initial onboarding wizard.
2. Your data can be retrieved by downgrading to an older version of Insomnia and exporting it.
3. Admittedly this is confusing, which is why we will be building another "Export" button in the "Preferences" and ship it as soon as possible.

## Can I still use Insomnia locally only?

Yes, you can via the Scratch Pad capability which is always local and doesn't require an account. You can also export your data and re-import a collection into Scratch Pad and keep working locally.

Also with Git Sync, the data is always stored in your Git repository and never in the cloud.

## Why is Insomnia making this move?

We are making this move for three reasons:

1) Roadmap: we are catering to new use-cases (collaboration, backups, synchronization, etc) that require the cloud to work and that many Insomnia users (not on this thread, _obviously_) are requiring.
2) Technical: we want to reduce the technical burden of having to support two backends for our storage, which has cause lots of problems in the past.
3) Business: Insomnia - whether is used locally or in the cloud - is a large project that requires significant resources to keep improving and getting better. By offering a cloud version of Insomnia, we can provide premium capabilities that can further allow us to keep this project running and active. This is not different from any other product that needs to find a way to self-sustain itself in the long term. The project also needs a lot more investments to do many of the things the community is asking, which means we need to allocate more resources, and we will. But we need a viable business to do all of these things.

## What about the security of my data in the cloud?

Everything that Insomnia does is end-to-end encrypted (E2EE) also when it's stored in the cloud. Nobody can access your data without a valid passphrase, which is why when using Insomnia is important to always remember the passphrase you have selected (which used to be associated to your account password in the past).Insomnia is built by Kong, which at the time of this writing works with 600+ enterprise organizations including Top Fortune 500 and Top Global 5000 around the world, so we can cater to every requirement from every security team in the world. Just get in touch with us.

On a personal note: we are trying to do the right thing for our users, many of which want us to offer capabilities that are not possible without a cloud. I understand that many of you are very upset by this change, which is why we are still committing to offering a local-only version of Insomnia via the Scratch Pad. The amount of effort it's taking us to offer a E2EE cloud is also outrageous, but we want to do the right thing and still provide it for all of our users free of charge (that is, also Free plan users).

I apologize for not having built a more intuitive way to export/import the data, this was an oversight as we were SO excited about all the other capabilities that we have shipped that we didn't think this through well enough. 

But it can still be done by:

* Downgrading to an older version of Insomnia
* Exporting the collection
* Re-importing the collection into Scratch Pad.

I hope that you will still find pleasure to use Insomnia whether it's in the cloud or locally in the Scratch Pad, but if this direction does not fit your ideas of where the project should go, then we will be sad to lose you.

**Comment by ptebault** [comment]
> But it can still be done by:
> 
> Downgrading to an older version of Insomnia
> Exporting the collection
> Re-importing the collection into Scratch Pad.

This should be an option in the wizard. I don't, can't, and won't keep my data in your cloud, for reasons.
Also, since the above flow was not an option visible in the wizard, the wizard is now gone.

Pull me with more features and cool stuff, fine, but don't take away the baseline local functionality. It's a social contract, and you've actively gained mindshare by engaging in that contract. You don't get to run off and choose a different path without incurring the cost of the loss of reputation, the loss of eyeballs, the loss of engagement, the loss of brand value.

Sorry to see you go.

**Comment by justin-caldicott** [comment]
I only need a simple rest client. Seems like the scratch pad may still be that option. But I'll probably look elsewhere because:
* Taking away data and making us export-import to get it back is not OK
* The main focus no longer seems to be a simple REST client
* I'm not sure how the scratch pad will work. If it was too much work to support a local data source, then surely the scratch pad is the same thing?

**Comment by subnetmarco** [comment]
Scratch Pad is easier on our end because it's only for collections (not design specs, tests, etc). It does not require an account, and it's local only.

It looks like this:

<img width="1728" alt="Screenshot 2023-09-27 at 6 15 51 PM" src="https://github.com/Kong/insomnia/assets/964813/eae05662-d5a5-4ed7-a369-efbe27a7a6e8">

**Comment by cha-king** [comment]
I'm all for the scratch pad, but let's not pretend that that banner isn't meant to be a deterrent 😁

**Comment by subnetmarco** [comment]
> Taking away data and making us export-import to get it back is not OK

There is a technical reason for this sub-par experience: we have users that are storing all sort of files (not just collections, but teams, many projects, tests, and so on). Given that Scratch Pad is limited and only support a collection, we cannot provide a quick migration to Scratch Pad for any users that doing more than just using Insomnia as a simple client.

By showing the Scratch Pad populated by default with only one collection, there would be a whole bunch of users thinking that all their other data is lost. So we are damned if we do and damned if we don't type of situation.

We opted to either "Export all data" in the onboarding wizard or "Move all to cloud" to keep it easy and not end up in weird data edge-cases with Scratch Pad.

**Comment by subnetmarco** [comment]
@cha-king We want to advertise that there is a lot more besides the Scratch Pad, even for Free users.

But in general, we built a local Scratch Pad, we subsidized Insomnia for 7 years, we subsidized E2EE for all users throughout all of this time, without basically monetizing any user (we didn't even have a login button in the app until recently). 

Please allow us to put a banner after almost a decade.

**Comment by platevoltage** [comment]
> @cha-king We want to advertise that there is a lot more besides the Scratch Pad, even for Free users.
> 
> But in general, we built a local Scratch Pad, we subsidized Insomnia for 7 years, we subsidized E2EE for all users throughout all of this time, without basically monetizing any user (we didn't even have a login button in the app until recently).
> 
> Please allow us to put a banner after almost a decade.

A dismiss button would be nice.

**Comment by TheBeachMaster** [comment]
No one asked for this. 👎🏿

**Comment by Arctomachine** [comment]
> @cha-king We want to advertise that there is a lot more besides the Scratch Pad, even for Free users.
> 
> But in general, we built a local Scratch Pad, we subsidized Insomnia for 7 years, we subsidized E2EE for all users throughout all of this time, without basically monetizing any user (we didn't even have a login button in the app until recently).
> 
> Please allow us to put a banner after almost a decade.

But here is the problem. If you want to monetize your userbase by inconveniencing them, you are not going to get far with that. Unity is doing something similar as we speak, by the way. You can check their situation and see where their not-well-thought decision got them.

**Comment by baek-sang** [comment]
Unfortunately, there's no reason to use insomnia anymore.
It forces cloud synchronization like postman does, but it has a worse UI, UX, and functionality than postman. If you need cloud sync, you're better off with postman (which also has a scratch pad).

alternative
https://github.com/usebruno/bruno
https://github.com/hoppscotch/hoppscotch
https://www.thunderclient.com/

**Comment by subnetmarco** [comment]
> It forces cloud synchronization like postman does

Scratch Pad is still here. It's local and doesn't require an account. You can still import your collection into Scratch Pad.

**Comment by crestonbunch** [comment]
@subnetmarco I understand as CTO you have to look at things from a business perspective, but I would encourage you to reflect deeper on what everyone in this thread is saying. Think about your product from a user perspective. If everyone is saying one thing, and you're saying another, you’re probably the one who is wrong. Learn from this mistake and do better next time. You lost a lot of goodwill in this thread.

**Comment by splix** [comment]
Insomnia just deleted all my existing requests, when it suggested to upgrade. It didn't ask to export, backup or anything before this upgrade. It's a total disaster.

**Comment by bertag** [comment]
> > It forces cloud synchronization like postman does
> 
> Scratch Pad is still here. It's local and doesn't require an account. You can still import your collection into Scratch Pad.

@subnetmarco Do plugins work with Scratch Pad?

**Comment by subnetmarco** [comment]
@splix For the sake of clarity:

> Insomnia just deleted all my existing requests

Your requests should still be there. You can either create an account with Insomnia (can be a Free account) or downgrade to a previous version of Insomnia. When creating an account, the data is end-to-end encrypted to the cloud vault.

> It didn't ask to export, backup or anything before this upgrade.

After the upgrade there is an onboarding wizard which says the data will be migrated to the cloud, and that it can be exported. Your data has not been migrated yet, unless an account has been created or a login has been executed.

**Comment by subnetmarco** [comment]
@bertag Not at the moment, but this was an oversight. Plugins will work with Scratch Pad with Insomnia 8.1 which will also include a more clear path way to export the data when in the Scratch Pad, for those that didn't export it during the onboarding wizard.

Insomnia 8.1 will ship soon, perhaps even this week.

**Comment by splix** [comment]
@subnetmarco no, it doesn't. The only option it provides is "Continue with ..." to migrate to the cloud and "Use Scratch Pad". There is no option to export or downgrade.

**Comment by subnetmarco** [comment]
@splix This is the screenshot of the wizard with title and description of what will happen, and the optionality of being able to export the data for portability. The data is also available if you downgrade, or available if you login.

With Git Sync, the data is never migrated to the cloud. With Scratch Pad, it is always local.

![image](https://github.com/Kong/insomnia/assets/964813/cc41e81b-7b8b-4bf9-b3f9-00dd3a6d1901)

**Comment by jeffreydwalter** [comment]
> What an amazing introduction to Kong's product offerings! I went from a happy user of Insomnia that was sharing how great it was with all of my colleagues, to a confused and upset online comment writer in just one short week! 🥳
> 
> Interestingly enough, I had just migrated to a new laptop and was overjoyed yesterday evening when I completed my development environment and closed my laptop for the night. I opened my laptop today to find that Insomnia has Amnesia and could not locate my data.
> 
> Needless to say...
> 
> N O T A G O O D L O O K
> 
> 👎🏻 👎🏻

Robbie! It's Jeff Walter, your former colleague. Nice to see a friend ranting in the GitHub issues. 😂

**Comment by splix** [comment]
@subnetmarco ok, I see it had a tiny action called "export your data", but once I clicked "Continue" it's not available anymore. There is no way back. It's not expected that it wouldn't give an access to your previous projects if you click that. 

It's a shady approach to convince users to migrate to the cloud by locking them in. You should be ashamed of doing that @subnetmarco. And obviously no one would trust Kong if they use such shady ways to manipulate people.

**Comment by jeffreydwalter** [comment]
> Given the large amount of feedback, I am writing here the current situation.
> 
> ## What changed with Insomnia 8.0 when it comes to your data?
> With Insomnia 8.0 we are doing two new things:
> 
> 1. After downloading the new binary (either manually or via the auto-update) we show an onboarding wizard that communicates that your data (collections, tests, etc) will be synchronized to the cloud after the user creates an account or signs in. There is an option to export the data, and the data is not actually synchronized until either an account is being created, or a login is being performed.
> 2. We are also introducing a new Scratch Pad mode that is always local and never synchronized to the cloud, and doesn't require an account with Insomnia.
> 
> The data is always end-to-end encrypted (E2EE) in the cloud, therefore no-one (not even Insomnia or Kong) can access it without a valid passphrase. Which is also why losing the passphrase can lead to loss of data, it's because of E2EE.
> 
> ## Can Insomnia still be used for Free?
> Yes, we offer a Free plan that you can use.
> 
> ## Why were these changes introduced?
> These changes were introduced because:
> 
> 1. We have a significant amount of the user base that wants to collaborate and automatically synchronize their data across their clients. This is impossible without leveraging the cloud to do so. Insomnia uses E2EE for each user, including the Free ones, to still be able to do this securely.
> 2. There is a significant effort in maintaining full support for an abstraction that supports two different backend (in the cloud, and locally), which is causing a large amount of issues and migration problems. We are therefore introducing limited support for the local backend storage in Scratch Pad only (more limited than the cloud one), and investing in making the cloud storage solid, stable and secure.
> 
> ## How to retrieve your data?
> 1. Your data can be retrieved in Insomnia 8.0 by exporting it via the initial onboarding wizard.
> 2. Your data can be retrieved by downgrading to an older version of Insomnia and exporting it.
> 3. Admittedly this is confusing, which is why we will be building another "Export" button in the "Preferences" and ship it as soon as possible.
> 
> ## Can I still use Insomnia locally only?
> Yes, you can via the Scratch Pad capability which is always local and doesn't require an account. You can also export your data and re-import a collection into Scratch Pad and keep working locally.
> 
> Also with Git Sync, the data is always stored in your Git repository and never in the cloud.
> 
> ## Why is Insomnia making this move?
> We are making this move for three reasons:
> 
> 1. Roadmap: we are catering to new use-cases (collaboration, backups, synchronization, etc) that require the cloud to work and that many Insomnia users (not on this thread, _obviously_) are requiring.
> 2. Technical: we want to reduce the technical burden of having to support two backends for our storage, which has cause lots of problems in the past.
> 3. Business: Insomnia - whether is used locally or in the cloud - is a large project that requires significant resources to keep improving and getting better. By offering a cloud version of Insomnia, we can provide premium capabilities that can further allow us to keep this project running and active. This is not different from any other product that needs to find a way to self-sustain itself in the long term. The project also needs a lot more investments to do many of the things the community is asking, which means we need to allocate more resources, and we will. But we need a viable business to do all of these things.
> 
> ## What about the security of my data in the cloud?
> Everything that Insomnia does is end-to-end encrypted (E2EE) also when it's stored in the cloud. Nobody can access your data without a valid passphrase, which is why when using Insomnia is important to always remember the passphrase you have selected (which used to be associated to your account password in the past).Insomnia is built by Kong, which at the time of this writing works with 600+ enterprise organizations including Top Fortune 500 and Top Global 5000 around the world, so we can cater to every requirement from every security team in the world. Just get in touch with us.
> 
> On a personal note: we are trying to do the right thing for our users, many of which want us to offer capabilities that are not possible without a cloud. I understand that many of you are very upset by this change, which is why we are still committing to offering a local-only version of Insomnia via the Scratch Pad. The amount of effort it's taking us to offer a E2EE cloud is also outrageous, but we want to do the right thing and still provide it for all of our users free of charge (that is, also Free plan users).
> 
> I apologize for not having built a more intuitive way to export/import the data, this was an oversight as we were SO excited about all the other capabilities that we have shipped that we didn't think this through well enough.
> 
> But it can still be done by:
> 
> * Downgrading to an older version of Insomnia
> * Exporting the collection
> * Re-importing the collection into Scratch Pad.
> 
> I hope that you will still find pleasure to use Insomnia whether it's in the cloud or locally in the Scratch Pad, but if this direction does not fit your ideas of where the project should go, then we will be sad to lose you.

Imprint into scratch pad didn't work for me.

**Comment by ineffyble** [comment]
This is incredibly short-sighted. Many organisations have migrated from Postman to Insomnia because of the Postman change, and now they're going to need to migrate again.

**Comment by subnetmarco** [comment]
I do agree that it's not clear how to export the data, so we will improve clarity by:

* Allowing to still export all data from the Scratch Pad view (so if someone misses the "export" button in the wizard, it can still be done later without having to downgrade).
* Re-enable plugins in Scratch Pad (this was an oversight, they will be enabled again).
* Making sure that import in Scratch Pad works in all scenarios (@jeffreydwalter if you have more data, please open a dedicated issue on GitHub so we can debug the problem).

This will ship son in the next v8.1, perhaps even this week (or early next week the latest). For any organization that requires strict security controls, you can get in touch with Kong/Insomnia and we will work with your security team to get the product approved even when in the cloud.

**Comment by ineffyble** [comment]
> For any organization that requires strict security controls, you can get in touch with Kong/Insomnia and we will work with your security team to get the product approved even when in the cloud.

Postman offers the same, but many organisations require employees to only used cloud tooling that has been through a vendor review and procurement process, which creates a significant barrier and prevents usage in the short-term.

**Comment by subnetmarco** [comment]
@ineffyble In this case, wouldn't the Insomnia Scratch Pad be that local-only option for debugging APIs without having to use the cloud?

**Comment by erraggy** [comment]
What a week, huh @subnetmarco ?
![image](https://github.com/Kong/insomnia/assets/366249/91ff4a09-6f3c-4aa7-b0d4-6901e84da93e)

**Comment by subnetmarco** [comment]
@erraggy Insomnia is not just a product, it's a way of life.

**Comment by ineffyble** [comment]
> @ineffyble In this case, wouldn't the Insomnia Scratch Pad be that local-only option for debugging APIs without having to use the cloud?

Hopefully so, I need to take a look at it. But one concern with Postman's equivalent (the "lightweight Postman API Client") has been the risk of a user accidentally switching to cloud without realising.

**Comment by subnetmarco** [comment]
@ineffyble I believe the risk of that happening to be low because creating an account is a very explicit multi-click operation that also redirects to the browser.

**Comment by Meggoo** [comment]
LBTM, I guess I'll use the humao rest client plugin now

**Comment by jeffreydwalter** [comment]
No need. I forked the repo and will stick with the old Insomnia until and
unless you guys decide to back away from your mistake. 👍 Clearly a lot of
people don't want this. Cheers.

**Comment by niksmac** [comment]
If you are using homebrew you can try `brew install --cask https://raw.githubusercontent.com/Homebrew/homebrew-cask/a649350f567a4ade1b6ee7509d798d6b1ab3a4fb/Casks/i/insomnia.rb` to install a downgraded version `v2023.5.8` which does not require a login. 

And I can confirm that I got all of my previous Projects back

**Comment by jdgarcia** [comment]
> The collection can also be re-imported into Scratch Pad, which will keep the data of the collection forever local without requiring an account.

Insomnia should just do this automatically upon updating. Without doing so, it does make it seem like it's lost your data. And doing this after you've updated and opened Scratch Pad means you have to manually downgrade in order to get the export option again.

**Comment by subnetmarco** [comment]
@jdgarcia I agree with you. We are making a release this week to make it more clear.

**Comment by chrisj-au** [comment]
I used to be a big proponent of insomnia but this is absolute shit.  Get stuffed Kong.

**Comment by CrystalENVT** [comment]
Massive L. Short term will use a Fork of Insomnia. Long term will be using a different client.

Trust forever broken by this shady & sudden change, trying to strong-arm users into having an account.

**Comment by backtozero** [comment]
It makes sense for everyone to search for alternatives. Even postman now makes sense again

**Comment by fizwidget** [comment]
I will also be migrating away from Insomina. How in earth did this update end up getting released? Absolutely mindboggling.

Introducing some extra nagging to create an account would’ve been slightly annoying but *fine*. Straight up locking people out of their data until they do? That is **so** not okay.

**Comment by Timwun** [comment]
This is really bad. Postman first, now Insomnia? I just don't want a Account...

**Comment by meiswjn** [comment]
> > It forces cloud synchronization like postman does
> 
> Scratch Pad is still here. It's local and doesn't require an account. You can still import your collection into Scratch Pad.

Will the scratch pad be available without an account in the future as well? Are there any gurantees regarding this?

**Comment by timharek** [comment]
There is Restfox, haven't tried it yet, but I'm going to give it a shot now:
https://github.com/flawiddsouza/Restfox

**Comment by 4ekki** [comment]
I was promoting Insomnia in my company for ~5 years. Good old days, :sigh:.

**Comment by ChristianSch** [comment]
Another data point: one of my devs didn't know that their data will be cloud-synced. So he accidentally leaked secrets. Great job, kong 👏  I just can't comprehend this fiasco

**Comment by CleverCoder** [comment]
same

**Comment by Aeolun** [comment]
I just don't understand how you could possibly have thought this was a good idea? What exactly did you think was the reason people used Insomnia instead of Postman? If I wanted everything could synced and shit I wouldn't have needed to migrate off Postman in the first place...

**Comment by CleverCoder** [comment]
> Time to fork.

Brilliant. Seriously, has someone created a fork yet?

It needs a new, clever name... something that respects the open source community and won't exploit the users. Honoring "keep it simple", and "do the right thing". Here's some ideas:
* Awakened
* RESTGuard
* TranquilREST
🤷

**Comment by lodi-g** [comment]
Hi @subnetmarco,

Good that you want to improve users' life. Not good that you force it onto everyone.

I won't repeat what everyone said (E2EE or not - I won't sign in), but rather add this: I contributed to Insomnia in the past and would have loved to contribute again. I'll not consider it anymore with this move.

**Comment by GeeDat** [comment]
I keep hearing that I can downgrade to a previous version of Insomnia and get my projects and environments back. How do I do that?

**Comment by lh0817** [comment]
Hey Guys,

Here a **downgrade instruction** as temporary workaround for the constellation **Ubuntu/Snap Insomnia package**:

```
sudo snap revert insomnia
sudo snap refresh --hold insomnia
```

The second command holds refreshes of insomnia indefinitely.

I hope it helps some people to be able to work today.

**Comment by alinmiron** [comment]
Stick the cloud up your arse! I was using Insomnnia because it was not forcing me to do  crap. Hope for a giant solar flare that would fk up all the "clouds".

**Comment by bcersows** [comment]
> I keep hearing that I can downgrade to a previous version of Insomnia and get my projects and environments back. How do I do that?

Depending on your system.

Here's some previous comments' instructions:
* Ubuntu: https://github.com/Kong/insomnia/issues/6577#issuecomment-1738650217
* Mac/Homebrew: https://github.com/Kong/insomnia/issues/6577#issuecomment-1738351378

Important to immediately disable auto-update after downgrading.

**Comment by krlz-dev** [comment]
Your are going to become postman soon :O and this was one of the main reasons I run from postman to Insomnia, to avoid all this nonsense, I lose many of the things I was working in this release, moving to https://hoppscotch.io/ for now

**Comment by Mark-Labuschagne** [comment]
Honestly, I moved from postman to Insomnia to have a simple and local API tester. Insomnia was perfect, I loved it. What a shame.

**Comment by tobias-feil-plana** [comment]
You should be ashamed of yourself. Insomnia was a trusted name for many developers in my friend circle and some of them were converted to Insomnia by me, being a nice alternative to Postman that doesn't suck your data. Now we're back to square one and an alternative like that doesn't exist anymore.

**Comment by Windowsfreak** [comment]
Insomnia is **DEAD** after this update.

Many people moved over from Postman, just to discover today that it's just its evil sister.

**Comment by mashaal** [comment]
<img width="313" alt="image" src="https://github.com/Kong/insomnia/assets/6100364/e36a6b8a-db60-403e-8253-2a519b3578cc">

oh fuck

**Comment by Rzpeg** [comment]
I will have to explore a self-hosted https://hoppscotch.io/ now.

**Comment by aniforprez** [comment]
Man how was the flow so poorly thought out? I've been working with some local collections for almost years now and all of that data just disappearing in the scratch pad almost gave me a heart attack. I made an account just so I could export my collections because the "wizard" doesn't let me go back to the export button and the scratch pad doesn't automatically import my collections that I NEVER synced to the cloud. Now my collections are taking up space on your servers just because of me being forced into this. I'm sure SOMEONE asked for this but then why was almost zero consideration given for local users before someone made this thread? Plugins not working was an oversight?

I'm sorry I'm adding to the dogpile but I'm thoroughly disappointed in this decision and will join the droves of people looking for alternatives. I didn't ask for this and now I'm forced to look at an ugly purple banner for being a plebian refusing to opt into the cloud "experience". This is very, very disheartening. Please know that all this complaining is cause people love Insomnia and want it to be good but this is a severe downgrade.

Edit: I'm finding out you cannot even delete your collections from the server through the website. You HAVE to log into the Insomnia desktop app and delete your collections from there. I cannot delete the automatically created organisation or my account from the website. My only recourse is to delete my account completely in the hope that it also deletes my organisation and, transitively, my collections. Automatically dumping everything on my private workspace to a cloud, encrypted or not, sure was a decision huh.

I cannot believe how this app has transitioned to *literally* be the same as Postman overnight. Just as Postman also deleted their scratchpad and thoroughly gimped their offline capabilities, I have no doubt this will be the path for Insomnia. I'm completely migrating off this app. I'm checking out [Bruno](https://www.usebruno.com/) if anyone's interested.

Edit 2: I just got an email saying I'm not a collaborator on my Personal Workspace. Not that the workspace was deleted. I'm assuming now my collections are just stuck on the cloud permanently. I created another account with the same email and password and now the default workspace doesn't have my collections. There's no way to delete a workspace despite me being the sole "collaborator" and owner of a workspace. The create account workflow is also busted and doesn't connect properly to the desktop app. It doesn't properly redirect the auth credentials to the app after you select a subscription which means I have to copy paste the login URL. This is just poor form all around folks.

**Comment by processor286** [comment]
Uninstalled Insomnia, am now giving Bruno a try, with a backup solution of Hurl.

**Comment by Windowsfreak** [comment]
1. Downgrade to 2023.5.7
2. Export
3. Try one of these
  - https://github.com/flawiddsouza/Restfox 
  - https://github.com/hoppscotch/hoppscotch
  - https://www.usebruno.com/

**Comment by baldurh** [comment]
Can someone clarify for me, if I upgrade to 8.0.0:

1. All my collections disappear?
2. I’m forced to sign up?
3. Will the collections appear again if I sign up?
4. If I sign up, are all my collections automatically uploaded to the cloud?
5. When collections are uploaded to the cloud, does that include the environment variables?

I’m mostly worried that at my workplace people will—without thinking—upgrade and sign up and all their collections—including environments—will be uploaded to the cloud.

**Comment by aniforprez** [comment]
> Can someone clarify for me, if I upgrade to 8.0.0:
> 
> 1. All my collections disappear?
> 2. I’m forced to sign up?
> 3. Will the collections appear again if I sign up?
> 4. If I sign up, are all my collections automatically uploaded to the cloud?
> 5. When collections are uploaded to the cloud, does that include the environment variables?
> 
> I’m mostly worried that at my workplace people will—without thinking—upgrade and sign up and all their collections—including environments—will be uploaded to the cloud.

1. As of writing this comment, all your collections are locked behind creating an account and are not moved to the scratchpad. Which means
2. Yes, you are forced to sign up to 
3. See your collections again because
4. They are automatically uploaded to the cloud
5. Which does, mercifully, include your env vars

Yes @baldurh, arguably the worst thing about this whole fiasco is all your data being dumped to the cloud without your consent. They claim it's all end-to-end-encrypted behind a password you're forced to create but I did not ask for this so it's a moot point

**Comment by baldurh** [comment]
> 1. As of writing this comment, all your collections are locked behind creating an account and are not moved to the scratchpad. Which means
> 2. Yes, you are forced to sign up to
> 3. See your collections again because
> 4. They are automatically uploaded to the cloud
> 5. Which does, mercifully, include your env vars
> 
> Yes @baldurh, arguably the worst thing about this whole fiasco is all your data being dumped to the cloud without your consent. They claim it's all end-to-end-encrypted behind a password you're forced to create but I did not ask for this so it's a moot point

Thanks @aniforprez!

**Comment by ghost** [comment]
I would only like to say a big thank you for this update, because I would never have discovered https://hurl.dev/ without this change.

**Comment by DylanDelobel** [comment]
I was going to export all my Postman collection to Insomnia,
Look like I will find another solution

**Comment by Ma-ve** [comment]
Much obliged turning Insomnia into ransomware! Ie., holding my data hostage unless I pay for it with my email address 👍.

**Comment by v1in** [comment]
I was so disappointed with today's update and overall such an accomplishment from Insomnia... I'll take another look at this client https://paw.cloud (now free)

**Comment by alfredoxperez** [comment]
Wow. I had a complex setup of requests. I heavily relied on Insomnia and was working to present this tool to a company for everyone to use with the setup I created, and now just gone. To say this situation is frustrating is an understatement.
